remove `format` from the Credential Response (#208)
openid/OpenID4VCI: openid-4-verifiable-credential-issuance-1_0.md

* `transaction_id`: OPTIONAL. String identifying a Deferred Issuance transaction. This claim is contained in the response if the Credential Issuer was unable to immediately issue the Credential. The value is subsequently used to obtain the respective Credential with the Deferred Credential Endpoint (see (#deferred-credential-issuance)). It MUST be present when the `credential` parameter is not returned. It MUST be invalidated after the Credential for which it was meant has been obtained by the Wallet.
* `c_nonce`: OPTIONAL. String containing a nonce to be used to create a proof of possession of key material when requesting a Credential (see (#credential_request)). When received, the Wallet MUST use this nonce value for its subsequent Credential Requests until the Credential Issuer provides a fresh nonce.
* `c_nonce_expires_in`: OPTIONAL. Number denoting the lifetime in seconds of the `c_nonce`.
* `notification_id`: OPTIONAL. String identifying an issued Credential that the Wallet includes in the Notification Request as defined in (#notification ). This parameter MUST NOT be present if `credential` parameter is not present.

> The Credential format and encoding of the `credential` MUST match the Credential format specified in the Credential Request. Details are defined in the Credential Format Profiles in (#format_profiles).

Reviewer: Encoding is not used anywhere else in this spec like this. I suggest to remove it.
```suggestion
The Credential format of the `credential` MUST match the Credential format specified in the Credential Request. Details are defined in the Credential Format Profiles in (#format_profiles).
```

Reviewer 2: we need to put somewhere in the spec that "The `format` parameter value determines the Credential format and encoding of the credential in the Credential Response.". maybe it does not belong here anymore, but it should not be just omitted like this.

Reviewer 2: ```suggestion
The Credential format and encoding of the `credential` is determined by the Credential format specified in the Credential Request. Details are defined in the Credential Format Profiles in (#format_profiles).
```

Author: I commited @Sakurann comment as it does not affect @awoie proposal to remove encoding. I think we need more input for keeping or removing encoding

Reviewer 3: I'm good with @Sakurann's new wording; it is accurate, the encoding of the credential does vary depending on the credential format (e.g. VCI requires that mdoc is base64url encoded in the response).

Reviewer: IMO, it is an unnecessary repetition but I'm ok with either way. To me that sentence is just a bit confusing since encoding is always implied by the format. There is no format identified by its format identifier with multiple encodings. If we have such a case, we should fix that.

Author: I actually agree with Oliver and only kept encoding in there to keep the changes as small is possible for this late PR

Reviewer 2: From the conversations I am having, it is not straightforward to the implementers that in the VP Token, sometimes credential is a string and sometimes it is an object. so we should be clear even if it feels redundant.

Reviewer: It is similar to the EdDSA problem in JOSE where it could mean Ed443 or Ed25519. We should define format identifiers unambiguously.

Reviewer: Also encoding is a characteristic of specific formats. Usually a format has one encoding. If there are legacy cases such as VCDM 1.1/1.0 where there are more encodings, we define already format identifiers like `jwt_vc_json`, and so on, that includes the encoding.

> The format and encoding of the `credential` value is determined by the Credential format specified in the Credential Request. Details are defined in the Credential Format Profiles in (#format_profiles).
> 
> Credential formats expressed as binary data MUST be base64url-encoded and returned as a string.

Author: ```suggestion
The format of the `credential` value is determined by the Credential format specified in the Credential Request. 

The encoding of the Credential in the `credential` value depends on the Credential Format Profiles defined in (#format_profiles). Credential formats expressed as binary data MUST be base64url-encoded and returned as a string.
```

Author: Is this an acceptable compromise? @Sakurann @awoie

Reviewer: To me that works. 

I think your suggestion makes it more clear that encoding refers to the encoding of the `credential` value and does not refer to the encoding of the credential itself. For instance, mdocs have a native CBOR encoding while being encoded as base64url in the `credential` value.

Reviewer 2: here is my suggestion

```suggestion
The format of the Credential in the Credential Response is determined by the value of the `format` parameter specified in the Credential Request.

The encoding of the Credential returned in the `credential` parameter depends on the Credential Format. Credential formats expressed as binary data MUST be base64url-encoded and returned as a string.

More details such as Credential Format identifiers are defined in the Credential Format Profiles in (#format_profiles). 
```
